### PR TITLE
chore: remove old navigation code [part 1]

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves 
https://artsyproduct.atlassian.net/browse/ONYX-1427

### Description
FINALLY 😭 

This PR is a first go at removing the old navigation code. Here I only remove the mentions of the feature flag and make the required changes - in a future PR I will follow-up with removing the old native module files as well.

I wanted initially to wait for 2 weeks before making this change but I figured out anyway it's something we want to release and if there's something urgent with the current release, we can anyway turn off the existing feature flag. I did though check Sentry and things seem fine. 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Remove old native code [part 1] - mounir

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Remove old native code [part 1] - mounir - MounirDhahri)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 30cfbef5bfe338b744f9c3a6da34e05e64454547
</p>

**Review by damassi — APPROVED** [review]
Screenshot 2024-12-04 at 7 29 59 AM

🔥 🔥 🔥

**Comment by MounirDhahri** [comment]
Sorry for dismissing your review @damassi - I just rebased

File: docs/best_practices.md [hunk 1]
```diff
@@ -326,9 +322,4 @@ The following parts of the iOS app are handled in native code:
 - Analytics for Native UI.
 - View In Room (Augmented Reality)
 - City Guide Drawer Handling
-
-The following parts of the iOS app are handled in native code, but will be managed by React Native once the new app shell is finished (expected March 2022):
-
-- ARScreenPresenterModule to navigate between view controllers.
-- The top-level tab bar, and each tab's navigation controller.
-- Deep-link and notification handling.
+  Ø
```
[the remark this answers is not in the document]
Comment (MounirDhahri) [reply docs/best_practices.md:325]: how did that sneak in there 😄 - great catch

**Pull request merged**